Source/Diagnostics/Diagnostics.cpp

```diff
@@ -199,29 +200,38 @@ Diagnostics::InitData ()
             InitializeFieldBufferData(i_buffer, lev);
         }
     }
-    // When particle buffers, m_particle_boundary_buffer are included,
-    // they will be initialized here
-    InitializeParticleBuffer();
 
     amrex::ParmParse pp_diag_name(m_diag_name);
+    // default for writing species output is 1
+    int write_species = 1;
+    pp_diag_name.query("write_species", write_species);
+    if (write_species == 1) {
+        // When particle buffers, m_particle_boundary_buffer are included,
+        // they will be initialized here
+        InitializeParticleBuffer();
+        InitializeParticleFunctors();
+    }
+
     amrex::Vector <amrex::Real> dummy_val(AMREX_SPACEDIM);
     if ( queryArrWithParser(pp_diag_name, "diag_lo", dummy_val, 0, AMREX_SPACEDIM) ||
          queryArrWithParser(pp_diag_name, "diag_hi", dummy_val, 0, AMREX_SPACEDIM) ) {
         // set geometry filter for particle-diags to true when the diagnostic domain-extent
-        // is specified by the user
-        for (int i = 0; i < m_output_species.size(); ++i) {
-            m_output_species[i].m_do_geom_filter = true;
+        // is specified by the user.
+        // Note that the filter is set for every ith snapshot, and the number of snapshots
+        // for full diagnostics is 1, while for BTD it is user-defined.
+        for (int i_buffer = 0; i_buffer < m_num_buffers; ++i_buffer ) {
+            for (int i = 0; i < m_output_species.size(); ++i) {
+                m_output_species[i_buffer][i].m_do_geom_filter = true;
+            }
+            // Disabling particle-io for reduced domain diagnostics by reducing
+            // the particle-diag vector to zero.
+            // This is a temporary fix until particle_buffer is supported in diagnostics.
+            m_output_species[i_buffer].clear();
```

Comment:
Here also now i_buffer is accessed instead of 0 before. For full diagnostics, we only have one buffer/snapshot. So we had a 0 in there. But the same would not have worked more generally for BTD. Thanks again